### PR TITLE
HARMONY-1901: Search for jobs by label

### PR DESCRIPTION
**Pull request by chris-durbin**
## Jira Issue ID
HARMONY-1901

## Description
Adds a query parameter `label` on the /jobs and /admin/jobs endpoint that allows filtering jobs based on the provided label(s). If multiple labels are provided the jobs returned will be a logical OR of all the provided labels (return all jobs that contain any of the labels).

## Local Test Steps
* Submit a few jobs with multiple labels in each job.
* Test /jobs?label=<comma separated list of labels> as well as /admin/jobs.
* Verify that jobs are filtered correctly based on the provided labels.
* Verify that each job returned has all the expected fields for the jobs listing endpoint and that the labels field contains all of the labels for each job (not just the label that was being searched for).

## PR Acceptance Checklist
* [X] Acceptance criteria met
* [X] Tests added/updated (if needed) and passing
* [X] Documentation updated (if needed)

### CODE REVIEW

File: services/harmony/app/models/job.ts [hunk 1]
```diff
@@ -500,16 +501,39 @@ export class Job extends DBRecord implements JobRecord {
...
-        .select(`${Job.table}.*`, tx.raw(`STRING_AGG(${LABELS_TABLE}.value, ',' order by value) AS label_values`))
-        .leftOuterJoin(`${JOBS_LABELS_TABLE}`, `${Job.table}.jobID`, '=', `${JOBS_LABELS_TABLE}.job_id`)
-        .leftOuterJoin(`${LABELS_TABLE}`, `${JOBS_LABELS_TABLE}.label_id`, '=', `${LABELS_TABLE}.id`)
-        .where(setTableNameForWhereClauses(Job.table, constraints.where))
-        .groupBy(`${Job.table}.id`)
-        .orderBy(
-          constraints?.orderBy?.field ?? 'createdAt',
-          constraints?.orderBy?.value ?? 'desc')
-        .modify((queryBuilder) => modifyQuery(queryBuilder, constraints));
+      if (constraints.labels) { // Requesting to limit the jobs based on the provided labels
```
Comment (indiejames) [inline services/harmony/app/models/job.ts:504]: Is it possible that someone might want to filter on labels, but doesn't need the labels included in the result? I think these are separate concerns.

**Review by indiejames — COMMENTED** [review]
Just a couple of comments.

File: services/harmony/app/models/job.ts (hunk 1 above)
Comment (chris-durbin) [reply services/harmony/app/models/job.ts:504]: There is currently not a use case for it or code that would request to not include labels but does filter on labels. If we do need that functionality at some point later we should add the code in then to handle that case too.

File: services/harmony/app/frontends/jobs.ts [hunk 2]
```diff
@@ -126,10 +127,17 @@ export async function getJobsListing(
     req.context.logger.info(`Get jobs listing for user ${req.user}`);
     const root = getRequestRoot(req);
     const { page, limit } = getPagingParams(req, env.defaultJobListPageSize);
+    let labels = [];
```
[the remark this answers is not in the document]
Comment (chris-durbin) [reply services/harmony/app/frontends/jobs.ts:130]: Nice, done.

**Review by indiejames — APPROVED** [review]
Tested  locally

File: services/harmony/app/models/job.ts (hunk 1 above)
Comment (vinnyinverso) [reply services/harmony/app/models/job.ts:504]: (Would need to add `includeLabels` as a parameter.)

**Review by vinnyinverso — APPROVED** [review]

**Pull request merged**